src/libs/json-ld.js

```diff
@@ -0,0 +1,150 @@
+import I18nText from 'shared_components/I18nText';
+
+export function getServiceJsonLdData(service, canonicalUrl) {
+  const category = I18nText.translate(service.categories[0].names);
+  switch (category) {
+    case 'Accommodation':
+      return getAccommodationJsonLd(service);
+    case 'Food':
+      return getFoodJsonLd(service);
+    case 'Activity':
+      return getActivityJsonLd(service, canonicalUrl);
+    default:
+      throw new Error('invalid category');
+  }
+}
+
+function getAccommodationJsonLd(service) {
+  const structuredData = {
+    '@context': 'https://schema.org/',
+    '@type': 'Hotel',
+    name: I18nText.translate(service.title),
+    image: getHeroImage(service),
+  };
+  if (service.description && I18nText.translate(service.description))
+    structuredData.description = I18nText.translate(service.description);
+  injectRatings(service, structuredData);
+  injectGeoCoordinates(service.originalLocation, structuredData);
```

Comment:
Also, I found that we were overwriting `service.location` but we were also saving the original location in `service.originalLocation`, thus using that.

Reply:
Cool! Merge it whenever you want